keymap.c

```diff
@@ -1298,6 +1301,152 @@ enum CommandResult mutt_parse_bind(struct Buffer *buf, struct Buffer *s,
+{
+  char *menu_names_dup = mutt_str_strdup(s);
+  char *menu_name = NULL;
+
+  while ((menu_name = strsep(&menu_names_dup, ",")))
+  {
+    int value = mutt_map_get_value(menu_name, Menus);
+    if (value == -1)
+    {
+      mutt_buffer_printf(err, _("%s: no such menu"), menu_name);
+      FREE(&menu_names_dup);
+      return NULL;
+    }
+    else
+      menu[value] = true;
+  }
+
+  FREE(&menu_names_dup);
+  return NULL;
+}
+
+/**
+ * km_unbind_all - Free all the keys in the supplied Keymap
+ * @param map  Keymap mapping
+ * @param mode Undo bind or macro, e.g. #MUTT_UNBIND, #MUTT_UNMACRO
+ *
+ * Iterate through Keymap and free keys defined either by "macro" or "bind".
+ */
+static void km_unbind_all(struct Keymap **map, unsigned long mode)
+{
+  struct Keymap *next = NULL;
+  struct Keymap *first = NULL;
+  struct Keymap *last = NULL;
+  struct Keymap *cur = *map;
+
+  while (cur)
+  {
+    next = cur->next;
+    if (((mode & MUTT_UNBIND) && !cur->macro) || ((mode & MUTT_UNMACRO) && cur->macro))
+    {
+      FREE(&cur->macro);
+      FREE(&cur->keys);
+      FREE(&cur->desc);
+      FREE(&cur);
+    }
+    else if (!first)
+    {
+      first = last = cur;
+    }
+    else if (last)
+    {
+      last->next = cur;
+      last = cur;
+    }
+    else
+    {
+      last = cur;
+    }
+    cur = next;
+  }
+
+  if (last)
+    last->next = NULL;
+
+  *map = first;
+}
+
+/**
+ * mutt_parse_unbind - Parse the 'unbind' command - Implements ::command_t
+ *
+ * Command unbinds:
+ * - one binding in one menu-name
+ * - one binding in all menu-names
+ * - all bindings in all menu-names
+ *
+ * unbind `<menu-name[,...]|*>` [`<key_sequence>`]
+ */
+enum CommandResult mutt_parse_unbind(struct Buffer *buf, struct Buffer *s,
+                                     unsigned long data, struct Buffer *err)
+{
+  int menu[MENU_MAX] = { 0 };
+  bool all_keys = false;
+  char *key = NULL;
+
+  mutt_extract_token(buf, s, 0);
+  if (mutt_str_strcmp(buf->data, "*") == 0)
+  {
+    for (int i = 0; i < MENU_MAX; i++)
+      menu[i] = 1;
+  }
+  else
+    parse_menu(menu, buf->data, err);
+
+  if (MoreArgs(s))
+  {
+    mutt_extract_token(buf, s, 0);
+    key = buf->data;
+  }
+  else
+    all_keys = true;
+
+  if (MoreArgs(s))
+  {
+    const char *cmd = (data & MUTT_UNMACRO) ? "unmacro" : "unbind";
+
+    mutt_buffer_printf(err, _("%s: too many arguments"), cmd);
+    return MUTT_CMD_ERROR;
+  }
+
+  for (int i = 0; i < MENU_MAX; i++)
+  {
+    if (menu[i] != 1)
+      continue;
+    if (all_keys)
+    {
+      km_unbind_all(&Keymaps[i], data);
+      km_bindkey("<enter>", MENU_GENERIC, OP_GENERIC_SELECT_ENTRY);
+      km_bindkey("<return>", MENU_GENERIC, OP_GENERIC_SELECT_ENTRY);
+      km_bindkey("<enter>", MENU_MAIN, OP_DISPLAY_MESSAGE);
+      km_bindkey("<return>", MENU_MAIN, OP_DISPLAY_MESSAGE);
+      km_bindkey("<backspace>", MENU_EDITOR, OP_EDITOR_BACKSPACE);
+      km_bindkey("\177", MENU_EDITOR, OP_EDITOR_BACKSPACE);
```

Comment:
What key does `\177` correspond to? Converting from octal to decimal to ASCII says it's Delete, but I don't see anything bound to delete after executing `unbind *`

Reply:
Yes, it's DEL character. It's for MENU_EDITOR otherwise it's not possible to delete characters in command prompt. At least in my case on MBP.

Reply:
After `:unbind *`, `:bind editor`, I see:
```
bind editor ^? backspace
bind editor <BackSpace> backspace
```

`^?` or <kbd>Ctrl-?</kbd> corresponds to ASCII `<Del>`.